packages/destination-actions/src/destinations/fullstory/index.ts

```diff
@@ -0,0 +1,56 @@
+import type { DestinationDefinition } from '@segment/actions-core'
+import { defaultValues } from '@segment/actions-core'
+import type { Settings } from './generated-types'
+import identifyUser from './identifyUser'
+import { listOperationsRequestParams, deleteUserRequestParams } from './request-params'
+import { dataRegions } from './data-regions'
+
+const destination: DestinationDefinition<Settings> = {
+  name: 'Fullstory (Actions)',
+  slug: 'actions-fullstory',
+  mode: 'cloud',
+  presets: [
+    {
+      name: 'Identify User',
+      subscribe: 'type = "identify"',
+      partnerAction: 'identifyUser',
+      mapping: defaultValues(identifyUser.fields)
+    }
+  ],
+  authentication: {
+    scheme: 'custom',
+    fields: {
+      apiKey: {
+        label: 'API Key',
+        description: '[FullStory API key](https://help.fullstory.com/hc/en-us/articles/360052021773-Managing-API-Keys)',
+        type: 'password',
+        required: true
+      },
+      region: {
+        label: 'Data Region',
+        description: 'The region where your FullStory organization is provisioned.',
+        type: 'string',
+        format: 'text',
+        choices: dataRegions,
+        default: 'north_america',
+        required: true
+      }
+    },
+
+    testAuthentication: (request, { settings }) => {
+      const { url, options } = listOperationsRequestParams(settings)
```

Comment:
List operations seemed like an inexpensive call to check if the auth configuration is valid that didn't require contextual params (e.g. an id).

Reply:
Oh wow, interesting. How is this being used in Segment?

Reply:
The `testAuthentication` function was scaffolded when creating the new destination with [Segment's Actions CLI](https://github.com/segmentio/action-destinations#actions-cli). I haven't been able to find any documentation or verify its use in local dev testing. My guess would be this is invoked when configuring the destination to verify the configured auth settings are valid. I'm not sure how I'd confirm this, though, until we get this work into private beta. Maybe it's worth reaching out to our Segment contacts?

This is the comment that scaffolded with the function:
```
      // Return a request that tests/validates the user's credentials.
      // If you do not have a way to validate the authentication fields safely,
      // you can remove the `testAuthentication` function, though discouraged.
```

Reply:
Ok, cool - yeah I was curious about frequency of access. Wondering if we would need to worry about it contributing to rate-limiting. Probably not if it's only called when the action is being configured in the Segment portal. Would be worth getting clarification from Segment folks. I don't think it should hold up your PR thought.

Reply:
@patrick-fs Closing the loop, we heard back from Segment that `testAuthentication` is called any time changes in settings are saved -- so this should be relatively infrequent.